rca/shortcourses/tests/test_models.py

```diff
@@ -0,0 +1,219 @@
+
+from django.test import TestCase
+
+from rca.home.models import HomePage
+from rca.programmes.models import ProgrammeType
+from rca.shortcourses.models import ShortCourseManualDate, ShortCoursePage
+
+from .test_access_planit import mocked_fetch_data_from_xml
+
+
+class TestBookingBarLogic(TestCase):
+    """ Test the various states that the booking bar logic can return, the logic
+    for this has become quite involved so the main aim of the testing here is
+    to describe how it acutall functions.
+
+    Scenarios:
+        1 - no data at all show applications are closed
+            - no page.show_register_link
+            - no manual dates
+            - no data from access planit
+        2 - If no booking data and show_register_link checked and AP id is preset
+            - populate auto register links in sidebar and in booking bar
+        3 - If no booking data and show_register_link checked and AP id is preset
+            and the manual_registration_url is defined populate manual register
+            links in sidebar and in booking bar
+        4 - If access planit course data comes through, but a page
+            application_form_url is defined, booking bar shows next AP course
+            date and "Submit for to apply". The line items in the modal should
+            use and show "apply" links in place of book links
+        5 - If manual booking dates are defined. The first/top booking date is
+            shown in the booking bar, 'Book' link in the booking bar opens modal
+            to show booking manually added booking items
+
+
+    """
+
+    def setUp(self):
+        ProgrammeType.objects.create(
+            display_name="Design", description="Some description text", id=1
+        )
+        self.home_page = HomePage.objects.first()
+        self.short_course_page = ShortCoursePage(
+            title="Short course title",
+            slug="short-course",
+            access_planit_course_id="1",
+            programme_type_id=1,
+            contact_url="https://rca.ac.uk",
+            contact_text="Read more",
+            hero_colour_option=1,
+            show_register_link=0,
+        )
+
+    def test_no_data(self):
+        """
+        1 no data at all show applications are closed
+            - no page.show_register_link
+            - no manual dates
+            - no data from access planit
+        """
+        self.home_page.add_child(instance=self.short_course_page)
+
+        # Get the auto register interest link made in the context
+        register_link = self.short_course_page.get_context(request=None)[
+            "register_interest_link"
+        ]
+
+        booking_bar_data = self.short_course_page._format_booking_bar(
+            register_interest_link=register_link, access_planit_data=None
+        )
+
+        self.assertEqual(
+            {
+                "message": "Applications are now closed",
+                "action": "Register your interest for upcoming dates",
```

Comment:
**nitpick**: This string is referenced in a few places, could it be a constant (for maintenance)